include/seqan3/core/detail/terminal.hpp

```diff
@@ -0,0 +1,86 @@
+//
+// Copyright (c) 2006-2017, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2017, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ * \brief Transfered from seqan2. Used in the argument parser to check console properties.
+ */
+
+#pragma once
+
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include <cstdio>
+
+#include <seqan3/core/platform.hpp>
+
+namespace seqan3::detail
+{
+
+// ----------------------------------------------------------------------------
+// Function is_terminal()
+// ----------------------------------------------------------------------------
+
+/*!\brief Check whether we are printing to a terminal.
+ *
+ * \return bool True if code is run in a terminal, False otherwise.
+ *               (Always false in Windows)
+ */
+inline bool is_terminal()
+{
+    return isatty(fileno(stdout));
+}
+
+// ----------------------------------------------------------------------------
+// Function get_terminal_size()
+// ----------------------------------------------------------------------------
+
+/*!\brief  Retrieve size of terminal.
+ *
+ * \return A pair [rows, cols] for the number of row and column bytes of the
+ *          current terminal.
```

Comment:
What means row and column bytes? Is it the number of...?

Reply:
Your function only returns the number of columns (=width). So this is misleading documentation.